src/components/YearSelector/index.tsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+import { ArrowBackIosNew, ArrowForwardIos } from '@mui/icons-material';
+import {
+  IconButton,
+  Button,
+  Typography,
+  Popover,
+  ListItemButton,
+} from '@mui/material';
+import { Dispatch, SetStateAction, useState } from 'react';
+
+interface YearSelectorProps {
+  selectedYear: number;
+  onChange: Dispatch<SetStateAction<number>>;
+}
+
+export default function YearSelector(props: YearSelectorProps) {
+  const MIN_YEAR = 2024;
+  const CURR_YEAR = new Date().getFullYear();
```

Comment:
this line causes this component not to be [pure](https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent) update this to be similar to the example in the docs (with a `useEffect`) also just a nit but `MIN_YEAR` can be moved outside of function so that we don't redefine it on every render. not a big deal tho